packages/features/bookings/Booker/components/OverlayCalendar/OverlayCalendarSettingsModal.tsx

```diff
@@ -0,0 +1,154 @@
+import Link from "next/link";
+import { useRouter } from "next/navigation";
+import { Fragment } from "react";
+
+import { classNames } from "@calcom/lib";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { trpc } from "@calcom/trpc/react";
+import {
+  Alert,
+  Dialog,
+  DialogContent,
+  EmptyScreen,
+  ListItem,
+  ListItemText,
+  ListItemTitle,
+  Switch,
+  DialogClose,
+  SkeletonContainer,
+  SkeletonText,
+} from "@calcom/ui";
+import { Calendar } from "@calcom/ui/components/icon";
+
+import { useLocalSet } from "../hooks/useLocalSet";
+import { useOverlayCalendarStore } from "./store";
+
+interface IOverlayCalendarContinueModalProps {
+  open?: boolean;
+  onClose?: (state: boolean) => void;
+}
+
+const SkeletonLoader = () => {
+  return (
+    <SkeletonContainer>
+      <div className="border-subtle mt-3 space-y-4 rounded-xl border px-4 py-4 ">
+        <SkeletonText className="h-4 w-full" />
+        <SkeletonText className="h-4 w-full" />
+        <SkeletonText className="h-4 w-full" />
+        <SkeletonText className="h-4 w-full" />
+      </div>
+    </SkeletonContainer>
+  );
+};
+
+export function OverlayCalendarSettingsModal(props: IOverlayCalendarContinueModalProps) {
+  const utils = trpc.useContext();
+  const setOverlayBusyDates = useOverlayCalendarStore((state) => state.setOverlayBusyDates);
+  const { data, isLoading } = trpc.viewer.connectedCalendars.useQuery(undefined, {
+    enabled: !!props.open,
+  });
+  const { toggleValue, hasItem } = useLocalSet<{
+    credentialId: number;
+    externalId: string;
+  }>("toggledConnectedCalendars", []);
+
+  const router = useRouter();
+  const { t } = useLocale();
+  return (
+    <>
+      <Dialog open={props.open} onOpenChange={props.onClose}>
+        <DialogContent
+          type="creation"
```

Comment:
pass enableOverflow prop to DialogContent. otherwise the content inside is overflowing when too many calendars are connected